src/completers/yarn.zsh

```diff
@@ -1,8 +1,19 @@
 #!/usr/bin/env zsh
 
 _fzf_complete_yarn() {
-    if [[ $@ = 'yarn workspace ' ]]; then
-        _fzf_complete_yarn-workspace '' $@
+    local subcommand=${${(Q)${(z)@}}[2]}
+
+    if [[ $subcommand = 'workspace' ]]; then
+        local workspace
+        if ! workspace=$(_fzf_complete_parse_argument 3 1 $@ '') && [[ -z $workspace ]]; then
```

Comment:
What does the last argument (`''`) mean?

Reply:
When that is deleted, Zsh displays `_fzf_complete_parse_argument:shift:5: shift count must be <= $#`.

Reply:
## TL;DR

To properly implement what have to be passed to `_fzf_complete_parse_argument`, store `$@` to a variable, then pass it. Passing the argument after that (`''`) is totally OK.

## Detailed Explanation

What's weird here is that the current parameters for `_fzf_complete_parse_argument` does not match the way it should. Firstly, let's take into account the spec of `_fzf_complete_parse_argument` that takes the following parameters:

- 1st: start index
- 2nd: index
- 3rd: arguments
- 4th: options that require arguments

However, `$@` implies it passes any number of arguments to that function. See what happens:

```zsh
show_params_num() {
    echo $#
}

show_params_directly() {
    show_params_num $@
}

show_params_indirectly() {
    local arguments=$@
    show_params_num $arguments
}

show_params_directly 'a' 'b' 'c'   # => 3
show_params_indirectly 'a' 'b' 'c' # => 1
```

When the number of arguments for `yarn workspace` exceeds 1, `_fzf_complete_parse_argument` would receive more than 4 arguments which are ignored.